src/Plugin/WebformElement/PaymentElement.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\os2forms_payment\Plugin\WebformElement;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\os2forms_payment\Helper\PaymentHelper;
+use Drupal\webform\Plugin\WebformElementBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'OS2forms_payment' element.
+ *
+ * @WebformElement(
+ *   id = "os2forms_payment",
+ *   label = @Translation("OS2forms payment element"),
+ *   description = @Translation("Provides a payment element."),
+ *   category = @Translation("Payment"),
+ * )
+ */
+class PaymentElement extends WebformElementBase {
+  /**
+   * Payment helper class.
+   *
+   * @var \Drupal\os2forms_payment\Helper\PaymentHelper
+   */
+  private PaymentHelper $paymentHelper;
+
+  /**
+   * {@inheritdoc}
+   *
+   * @phpstan-param array<string, mixed> $configuration
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->paymentHelper = $container->get(PaymentHelper::class);
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @return array<mixed>
+   *   Returns array of default properties.
+   */
+  protected function defineDefaultProperties() {
+    return [
+      'amount_to_pay' => '',
+      'checkout_page_description' => '',
+    ] + parent::defineDefaultProperties();
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param array<mixed> $form
+   *   Form object.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state object.
+   *
+   * @return array<mixed>
+   *   Returns modified form object.
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+    $form = parent::form($form, $form_state);
+    $form['element']['amount_to_pay'] = [
+      '#type' => 'textfield',
+      '#title' => $this
+        ->t('Machine name of field containing amount to pay'),
+    ];
```

Comment:
Let the user select the element rather than having it enter a key. See https://github.com/itk-dev/os2forms_digital_post/blob/main/src/Plugin/WebformHandler/WebformHandlerSF1601.php#L107-L114 for hints on how to do this.

Reply:
Cool. Got it!